sha256.go

```diff
@@ -0,0 +1,66 @@
+/*
```

Comment:
why is the synopsis in this file?

Reply:
godoc concat all files together and doc header from some file use as header... is possible to extract this to some special file like `doc.go`. maybe it's more comprehensible? (like https://golang.org/src/fmt/doc.go)